### PR TITLE
Improve test case attributes

### PR DESCRIPTION
**Pull request by nathanwoctopusdeploy**
# Background

Improve test case attributes

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Halibut.Tests/Support/TestAttributes/LatestClientAndLatestServiceTestCasesAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+namespace Halibut.Tests.Support.TestAttributes
+{
+    /// <summary>
+    ///     Holds all the standard test cases for testing a latest client with a latest service.
+    ///     In this case latest means the code as is, rather than some previously built version of halibut.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = true, Inherited = false)]
+    public class LatestClientAndLatestServiceTestCasesAttribute : TestCaseSourceAttribute
+    {
+        public LatestClientAndLatestServiceTestCasesAttribute(bool testWebSocket = true, bool testNetworkConditions = true) :
```
Comment (LukeButters) [inline source/Halibut.Tests/Support/TestAttributes/LatestClientAndLatestServiceTestCasesAttribute.cs:15]: nit: I wonder if network conditions could be an enum for description e.g.
```
TestNetworkCondtions : Network.All
TestNetworkCondtions : Network.PerfectOnly
```

maybe the bool reads better actually...

**Comment by nathanwoctopusdeploy** [comment]
[sc-53455]

File: source/Halibut.Tests/Support/TestAttributes/TestCasesAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using NUnit.Framework;
+
+namespace Halibut.Tests.Support.TestAttributes
+{
+    /// <summary>
+    /// Indicates the source to be used to provide test fixture instances for a test class.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = true, Inherited = false)]
+    public class TestCasesAttribute : TestCaseSourceAttribute
```
Comment (nathanwoctopusdeploy) [inline source/Halibut.Tests/Support/TestAttributes/TestCasesAttribute.cs:10]: Unused, to be removed.

File: source/Halibut.Tests/Support/TestAttributes/LatestClientAndLatestServiceTestCasesAttribute.cs (hunk 1 above)
Comment (acodrington) [reply source/Halibut.Tests/Support/TestAttributes/LatestClientAndLatestServiceTestCasesAttribute.cs:15]: Yeah I like the bool tbh - we're either testing network conditions, or the network is perfect. Either/Or, nice and simple

**Review by acodrington — APPROVED** [review]
Love it, no notes, 10/10, would review again!

**Pull request merged**